### PR TITLE
Add `Maybe` Documentation

### PR DESCRIPTION
**Pull request by evilsoft**
## This is good...Maybe?
image

This PR adds the README documentation for the `Maybe` type and all of it's support functions. Also updated some issues with some of the siggys.

Once this is merged, there will follow another PR to move this into the documents.

This goes against [this issue](https://github.com/evilsoft/crocks/issues/42)

### CODE REVIEW

File: src/Maybe/README.md [hunk 1]
```diff
@@ -0,0 +1,1321 @@
...
+//=> Just 32
+
+Just('hello')
+  .map(joinWith(' -- '))
+  .ap(Just('friend'))
+//=> Just "hello -- friend"
+
+Maybe.of(add)
+  .ap(Just(29))
+  .ap(Nothing())
```
Comment (foxdonut) [inline src/Maybe/README.md]: `//=> Nothing`

File: src/Maybe/README.md [hunk 2]
```diff
@@ -0,0 +1,1321 @@
...
+Maybe a = Nothing | Just a
+```
+
+Defined as a Sum Type with it's Left side fixed to `()` (`Nothing`), `Maybe` is
+well suited for capturing disjunction when the cause of the "error" case does
+not need to be communicated. For example, providing default values on specific
+conditions.
+
+A `Maybe` represents disjunction by using (2) constructors, `Nothing` or `Just`.
+A `Just` instance represents the truth case while, `Nothing` is considered
```
Comment (rpearce) [inline src/Maybe/README.md]: methinks you don't need a comma here

File: src/Maybe/README.md [hunk 3]
```diff
@@ -0,0 +1,1321 @@
...
+
+#### Just
+
+```haskell
+Maybe.Just :: a -> Maybe a
+```
+
+Used to construct a `Just` instance that represents the "true" portion of a
+disjunction or a valid value. `Just` will wrap any given value in a `Just`,
+which signals the validity of the wrapped value.
```
Comment (rpearce) [inline src/Maybe/README.md]: Another comma thing... how do either of these work for ya:
```
`Just` will wrap any given value in a `Just`, and this signals the validity of the wrapped value.
```
or
```
`Just` will wrap any given value in a `Just`, signalling the validity of the wrapped value.
```

File: src/Maybe/README.md [hunk 4]
```diff
@@ -0,0 +1,1321 @@
...
+```
+
+#### of
+
+```haskell
+Maybe.of :: a -> Maybe a
+```
+
+Used to wrap any value into a `Maybe` as a `Just`, `of` is used mostly by
+helper functions that work "generically" with `Applicative` or `Monad` types
```
Comment (rpearce) [inline src/Maybe/README.md]: "types" should probs be "type" as "instances" covers the plurality?

File: src/Maybe/README.md (hunk 3 above)
Comment (evilsoft) [reply src/Maybe/README.md]: I :heart: the latter.

File: src/Maybe/README.md (hunk 4 above)
Comment (evilsoft) [reply src/Maybe/README.md]: Actually think we can :hocho: the word `type` altogether.

File: src/Maybe/README.md (hunk 1 above)
Comment (evilsoft) [reply src/Maybe/README.md]: Addressed.

**Comment by evilsoft** [comment]
image

**Pull request merged**